### PR TITLE
dbeaver/pro#2569 [ERD] Rearrange of connections for diagram should be consistent.

### PR DESCRIPTION
**Pull request by serjiokov**
Fixed entity ordering for layout representation;
order of connection for draw;
improved logic of intersection;
some broken connections;

etc.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java [hunk 1]
```diff
@@ -61,6 +89,28 @@ public void visit(DirectedGraph graph) {
         drawMissedNodes(isolatedNodes, nodeMissed, nodeByLevels);
     }
 
+    private int computeDistanceX(List<Node> nodes) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java]:
```suggestion
    private int computeDistanceX(@NotNull List<Node> nodes) {
```

File: plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java [hunk 2]
```diff
@@ -61,6 +89,28 @@ public void visit(DirectedGraph graph) {
...
+        if (maxCountOfEdges > 0) {
+            distance += maxCountOfEdges * DISTANCE_PER_EDGE_X;
+        }
+        if (distance < DISTANCE_PER_EDGE_X) {
+            distance = DISTANCE_ENTITIES_X;
+        }
+        return distance;
+    }
+
+    private int computeDistanceY(Node n) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java:123]:
```suggestion
    private int computeDistanceY(@NotNull Node n) {
```

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java [hunk 3]
```diff
@@ -322,7 +411,8 @@ private void createGraphLayers(TreeMap<Integer, List<Node>> nodeByEdges, int idx
         duplicationNode2index.forEach((key, value) -> nodeByEdges.get(value).remove(key));
     }
 
-    private Integer getNodeIndex(TreeMap<Integer, List<Node>> nodeByEdges, Node src) {
+    @Nullable
+    private Integer getNodeIndex(@NotNull TreeMap<Integer, List<Node>> nodeByEdges,@NotNull Node src) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/layout/algorithm/direct/OrthoDirectedGraphLayout.java]:
```suggestion
    private Integer getNodeIndex(@NotNull TreeMap<Integer, List<Node>> nodeByEdges, @NotNull Node src) {
```

**Comment by MashaKorax** [comment]
verified

**Pull request merged**